src/preferences/configobject.cpp

```diff
@@ -186,34 +187,66 @@ template <class ValueType> void ConfigObject<ValueType>::reopen(const QString& f
     }
 }
 
-template <class ValueType> void ConfigObject<ValueType>::save() {
+/// Save the ConfigObject to disk.
+/// Returns true on success
+template<class ValueType>
+bool ConfigObject<ValueType>::save() {
     QReadLocker lock(&m_valuesLock); // we only read the m_values here.
-    QFile file(m_filename);
-    if (!QDir(QFileInfo(file).absolutePath()).exists()) {
-        QDir().mkpath(QFileInfo(file).absolutePath());
+    QFile tmpFile(m_filename + kTempFilenameExtension);
+    if (!QDir(QFileInfo(tmpFile).absolutePath()).exists()) {
+        QDir().mkpath(QFileInfo(tmpFile).absolutePath());
+    }
+    if (!tmpFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        qWarning() << "Could not write config file: " << tmpFile.fileName();
+        return false;
+    }
+    QTextStream stream(&tmpFile);
+    stream.setCodec("UTF-8");
+
+    QString group = "";
+
+    uint minLength = 0;
```

Comment:
signed/unsigned mismatch, use `qint64` like pos()

I wonder why you didn't notice any warnings or build failures using -DWARNINGS_FATAL=ON as suggested?

Reply:
since floating convresion got a warning this is not possible anymore. my env did not print any type warning otherwise I would have changed it.

Reply:
> since floating convresion got a warning this is not possible anymore.

What do you mean? Are there still float-conversion warnings in the code? Latest 2.3/main should be clean, you can `git pull` the changes into this branch to get all fixes from upstream.

Reply:
I merged it before. Did you disable ccache ? I get at least a dozen warnings when it has to compile many files.

Reply:
https://gist.github.com/poelzi/a1acac5d0a2109c5b6ed890cd422cd18

Reply:
Weird, what compiler + version are you using? Yes, I disabled ccache and get no warnings with gcc 10.2.0. Also, our CI builds also don't warn about these conversions even though they have `WARNINGS_FATAL` enabled.

I'll look into fixing these errors.

Reply:
I currently compile with clang version 7.1.0 (tags/RELEASE_710/final)

Reply:
I'll try to set up clang, but our CI is using clang as well and doesn't complain.

I've gone through your log and hopefully fixed all warnings: https://github.com/mixxxdj/mixxx/pull/3227

Please test.

Reply:
Compiling with Clang 10 did not show any new errors besides #3152.

Reply:
Strange, newer compilers often are more strict than old versions.

Reply:
@poelzi Why did you mark this as resolved?

Reply:
because I forgot to push :(